devices/ePaper/Drivers/Ssd1681.cs

```diff
@@ -0,0 +1,718 @@
+                Mode = SpiMode.Mode0,
+                ChipSelectLineActiveState = false,
+                Configuration = SpiBusConfiguration.HalfDuplex,
+                DataFlow = DataFlow.MsbFirst
+            };
+
+            this.spiDevice = new SpiDevice(spiConnectionSettings);
+
+            // setup the gpio pins
+            this.resetPin = gpioController.OpenPin(resetPin, PinMode.Output);
+            this.dataCommandPin = gpioController.OpenPin(dataCommandPin, PinMode.Output);
+            this.busyPin = gpioController.OpenPin(busyPin, PinMode.Input);
+
+            this.Width = width;
+            this.Height = height;
+            PowerState = PowerState.Unknown;
+
+            this.InitializeFrameBuffer(width, height, enableFramePaging);
+            this.CalculateFrameBufferPageBounds();
+        }
+
+
+        /// <inheritdoc/>
+        public int Width { get; }
+
+        /// <inheritdoc/>
+        public int Height { get; }
+
+        /// <inheritdoc/>
+        public IFrameBuffer FrameBuffer
+            => this.frameBuffer2bpp;
+
+        /// <summary>
+        /// Gets the current power state of the display panel.
+        /// </summary>
+        public PowerState PowerState { get; private set; }
+
+
+
+        /// <summary>
+        /// Performs the required steps to "power on" the display.
+        /// </summary>
+        public void PowerOn()
+        {
+            this.HardwareReset();
+            this.SoftwareReset();
+
+            PowerState = PowerState.PoweredOn;
+        }
+
+        /// <summary>
+        /// Puts the display to sleep using the specified <see cref="SleepMode"/>.
+        /// </summary>
+        /// <param name="sleepMode">The sleep mode to use when powering down the display.</param>
+        public void PowerDown(SleepMode sleepMode = SleepMode.Normal)
+        {
+            this.SendCommand((byte)Command.DeepSleepMode);
+            this.SendData((byte)sleepMode);
+
+            PowerState = PowerState.PoweredOff;
+        }
+
+        /// <summary>
+        /// Perform the required initialization steps to set up the display.
+        /// </summary>
+        public void Initialize()
+        {
+            if (PowerState != PowerState.PoweredOn)
+                throw new InvalidOperationException("Unable to initialize the display until it has been powered on. Call PowerOn() first.");
+
+            // set gate lines and scanning sequence
+            this.SendCommand((byte)Command.DriverOutputControl);
+            this.SendData((byte)(this.Height - 1), 0x00, 0x00); // refer to spec for description
+
+            // Set data entry sequence
+            this.SendCommand((byte)Command.DataEntryModeSetting);
+            this.SendData(0x03); // Y Increment, X Increment with RAM address counter auto updated in the X direction.
+
+            // Set RAM X start / end position
+            this.SendCommand((byte)Command.SetRAMAddressXStartEndPosition);
+            this.SendData(
```

Comment:
I would prefer to have the comment out of the line. We're mainly using it before the line and not in the instructions